src/fileio/get_s3_endpoint.cpp

```diff
@@ -15,15 +16,32 @@ namespace fileio {
 // Endpoint addresses specified on http://docs.aws.amazon.com/general/latest/gr/rande.html
 const std::vector<std::string> AWS_S3_END_POINTS {
   "s3.amazonaws.com",
-  "s3-us-west-2.amazonaws.com",
   "s3-us-west-1.amazonaws.com",
+  "s3-us-west-2.amazonaws.com",
   "s3-eu-west-1.amazonaws.com",
+  "s3-eu-central-1.amazonaws.com",
   "s3-ap-southeast-1.amazonaws.com",
   "s3-ap-southeast-2.amazonaws.com",
   "s3-ap-northeast-1.amazonaws.com",
+  "s3-ap-northeast-2.amazonaws.com",
   "s3-sa-east-1.amazonaws.com",
+  "s3-ap-south-1.amazonaws.com",
 };
 
+const std::map<std::string, std::string> AWS_S3_ENDPOINT_TO_REGION {
```

Comment:
It looks like there are other S3 regions. Do we want to add more here?

http://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region